odl/solvers/advanced/proximal_operators.py

```diff
@@ -143,26 +251,28 @@ def make_prox(tau):
 
         return IdentityOperator(space)
 
-    return make_prox
+    return make_identity
 
 
-def proximal_nonnegativity(space):
-    """Function to create the proximal operator of G(x) = ind(x > 0).
+def proximal_clamp(space, lower=None, upper=None):
```

Comment:
Clamp is the standard word in programming, see e.g.

http://http.developer.nvidia.com/Cg/clamp.html

With that said, "clamping" is the operation performed, not the underlying constraint/function. We have named all others after this so i'll go for your `proximal_box_constraint` here.